### ISSUE TITLE
Better error message for `Query.count` out of bounds

### ISSUE THREAD

**Issue by rtfeldman** [eeue56/elm-html-test#59]
Currently:

```
    ▼ Query.index 1

    0 matches found for this query.


    ✗ Query.index 1 always expects to find 1 element, but it found 0 instead.
```

Better:

```
    ▼ Query.index 1

    0 matches found for this query.


    ✗ Query.index 1 returned no matches because it was searching a list of length 1. The highest index that could find an element in this list would be Query.index 0.
```

**Comment by eeue56** [comment]
Proposed better is too wordy I think. How about something alone the lines

> Query.index 1 expects a list of two elements, but I got one
